libmscore/note.cpp

```diff
@@ -1530,6 +1530,13 @@ Element* Note::drop(const DropData& data)
                         delete e;
                         return 0;
                         }
+                  if (e->staffIdx() != staffIdx()) {
```

Comment:
Hmm, good question!  I will take another look, probably I could remove the test for staffIdx and just execute this code always.